incubator/hnc/internal/validators/object.go

```diff
@@ -275,16 +276,19 @@ func (o *Object) handleInherited(op k8sadm.Operation, newSource, oldSource strin
 	return deny(metav1.StatusReasonInternalError, "unknown operation: "+string(op))
 }
 
-// isDeletingNS returns true if the namespace of the object is already being deleted, based on the
-// in-memory forest.
-func (o *Object) isDeletingNS(inst *unstructured.Unstructured) bool {
-	o.Forest.Lock()
-	defer o.Forest.Unlock()
-	ns := o.Forest.Get(inst.GetNamespace())
-	if ns == nil {
-		return false
+// validateDeletingNS validates if the namespace of the object is already being deleted
+func (o *Object) validateDeletingNS(ctx context.Context, inst *unstructured.Unstructured) error {
+	ns := &corev1.Namespace{}
+	nnm := types.NamespacedName{Name: inst.GetNamespace()}
+	if err := o.client.Get(ctx, nnm, ns); err != nil {
+		return err
 	}
-	return ns.IsDeleting
```

Comment:
Can you please remove all the code that _sets_ `ns.IsDeleting` as well, assuming this was the only place that read it?

Reply:
can you help to check whether these [lines](https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/internal/reconcilers/hierarchy_config.go#L295-L307)  should be removed? im not quite sure what its doing.

Reply:
will re-push after this get confirmed

Reply:
Those lines are still needed, but you can just check `nsInst.DeletionTimestamp.IsZero()` instead (check out line 220; this function is called on line 228).

Reply:
... fyi, what it's doing is creating a subnamespace. Here's the flow:

1. User creates a `SubnamespaceAnchor` in the parent
2. HNC's anchor reconciler notices the anchor in the parent, and creates the subnamespace with an annotation pointing to the parent, but it doesn't create the `HierarchyConfiguration` object pointing back to the parent. This allows the anchor reconciler to be atomic (e.g. it only needs to create one object in a single call).
3. HNC's hierarchy config reconciler notices the namespace with the annotation, and updates the `HierarchyConfiguration` to match it.

This code is trying to suppress step 3 from happening when the namespace is being deleted.

Reply:
got it